### PR TITLE
FAS-18-signup-bug resolved

### PR DESCRIPTION
**Pull request by theninjacoder-uz**

### CODE REVIEW

File: src/api_gateway/handlers/customer_handler/customer_handler.go [hunk 1]
```diff
@@ -51,25 +49,42 @@ func (ch *customerHandler) TestHandler2(w http.ResponseWriter, r *http.Request)
...
-			u.WriteJSON(w, "one of the fields is incorrect")
-		}
+		utils.HandleInternalWithMessage(w, err, "error in reading from redis")
+		return
+	}
+
+	// if the login value is incorrect or wasn't set before throw an error
+	v, ok := i.([]byte)
+	if !ok || string(v) == "" {
+		utils.HandleBadRequestResponse(w, "login value is incorrect or time out")
```
Comment (khdoba2000) [inline src/api_gateway/handlers/customer_handler/customer_handler.go:75]: why login?

File: src/api_gateway/handlers/customer_handler/customer_handler.go [hunk 2]
```diff
@@ -229,14 +242,34 @@ func (ch *customerHandler) VerifyCodeHandler(w http.ResponseWriter, r *http.Requ
...
+		}
+
 		w.WriteHeader(http.StatusOK)
 		u.WriteJSON(w, models.LoginResponse{
 			IsAuthenticated: false,
 		})
 		return
 	}
+
+	// other actions (ex: phone recover) shorter timeout also need to be set
```
Comment (khdoba2000) [inline src/api_gateway/handlers/customer_handler/customer_handler.go:267]: why you need to store verified True status while in just lower lines you are returning jwt tokens?

Comment (theninjacoder-uz) [reply src/api_gateway/handlers/customer_handler/customer_handler.go:267]: IsAuthenticated field added to inform front signUP required. Ok, I can remove this field and return empty struct with StatusOk then front knows  signUP required if it recieves empty response.

File: src/api_gateway/handlers/customer_handler/customer_handler.go (hunk 1 above)
Comment (theninjacoder-uz) [reply src/api_gateway/handlers/customer_handler/customer_handler.go:75]: It's changed in FAS-22 task, I made a basic method for verifying email or phone

File: src/api_gateway/handlers/customer_handler/customer_handler.go (hunk 2 above)
Comment (khdoba2000) [reply src/api_gateway/handlers/customer_handler/customer_handler.go:267]: I mean why to store verified true to redis?

Comment (khdoba2000) [reply src/api_gateway/handlers/customer_handler/customer_handler.go:267]: You can remove storing to redis when It is not any Reset or Signup action.

**Pull request merged**